### DISCUSSION TITLE
CSV import related issues

### DISCUSSION THREAD

**Discussion by letwang** [PrestaShop/PrestaShop#34978] [Q&A]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Advanced Parameters -> Import
For example, if you select "Products" to import, the "Available fields" on the right and the "Download sample csv files" on the lower right do not completely correspond.
# Optional and optional default values are not marked, and required items are not marked.
Which items can be ignored, and what the default value is if ignored, are currently not clearly marked, and all required fields are not completely marked.
# Categories (x,y,z...) Feature (Name:Value:Position:Customized)
I don't know how to use these, can I only enter ID, or can I also enter text? There is no demonstration in the example, so I don't know how to use it. Please add a complete CSV sample file.

# Import -> Match your data
There is no correct and complete automatic correspondence.
```
fputcsv($fp, ['ID', 'Active (0/1)', 'Name*', 'Categories (x,y,z...)', 'Price tax excluded', 'Price tax included', 'On sale (0/1)', 'Discount percent', 'Discount from (yyyy-mm-dd)', 'Discount to (yyyy-mm-dd)', 'Reference #', 'Supplier', 'Brand', 'Quantity', 'Minimal quantity', 'Visibility', 'Summary', 'Description', 'Tags (x,y,z...)', 'Meta title', 'Meta keywords', 'Meta description', 'Available for order (0 = No, 1 = Yes)', 'Show price (0 = No, 1 = Yes)', 'Image URLs (x,y,z...)', 'Delete existing images (0 = No, 1 = Yes)', 'Feature (Name:Value:Position:Customized)']);
```
Even though I have generated the CSV exactly according to "Available fields", when it comes to "Match your data", there are still some data corresponding errors.

### Expected behavior

Looking forward to CSV import being simpler and easier to use.
It's currently too complex, too difficult to understand, there are no adequate and perfect demonstrations, and it's too time-consuming.

### Steps to reproduce

1.Advanced Parameters -> Import
2.Select "Products" to import
3.Match your data

### PrestaShop version(s) where the bug happened

8.1.3

### PHP version(s) where the bug happened

7.4

**Comment by AureRita** [comment]
Hi @letwang 

Thank you for your report, as I understand, you want to improve the system of the Import, this one could be interpreted like some new feature you want to import for prestashop ? 

Are you agree with that ? If yes, I'll transform this one on discusion to give your opinion to our Product council that can give this to our dev team 

Waiting for your feedback

**Comment by letwang** [comment]
OK. Please make your own arrangements and I respect your work pace.

**Comment by AureRita** [comment]
FYI @PrestaShop/product-council

**Comment by fulldecent** [comment]
Here is a specific part of the product import process that is broken. Reported separately as a bug. If you use the example CSV file and try to import that it fails. Video included. https://github.com/PrestaShop/PrestaShop/issues/36117

**Comment by fulldecent** [comment]
Feature request: add export button here.

Adding a direct EXPORT button here will be helpful. (Even if it is possible to export using some other method.)

This is because people will see the symmetry of the EXPORT and the IMPORT buttons. Then they know that they can export, fix things in Excel, and then import.

---

Also, this will help with the maintenance of PrestaShop. Because [right now the example import files are incorrect](https://github.com/PrestaShop/PrestaShop/issues/36117). If these could be generated with the EXPORT button then contributors to the PrestaShop project can easily create new example files automatically rather than manually.

**Reply by fulldecent** [reply]
Related: https://www.prestashop.com/forums/topic/1025672-export-products-full-export-with-all-fields/
